File: go/binary-tree-inorder-traversal/binary-tree-inorder-traversal.go
package main

import(
	"fmt"
)

type TreeNode struct {
    Val int
    Left *TreeNode
    Right *TreeNode
}

func inorderTraversal (root *TreeNode) []int{
	ret := []int{}

    var dfs func(root *TreeNode)
    dfs = func(root *TreeNode){
        if root == nil {
            return 
        }
        dfs(root.Left)
        ret = append(ret, root.Val)
        dfs(root.Right)
    }
    dfs(root)
    return ret
}

func main(){
    root := &TreeNode{Val: 1}
    nodeB := &TreeNode{Val: 2}
    nodeC := &TreeNode{Val: 3}

    root.Right = nodeB
    nodeB.Left = nodeC

	res := inorderTraversal(root)

	fmt.Println("answer to the first example is:", res)
}

File: js/binary-tree-inorder-traversal/binary-tree-inorder-traversal.js
function TreeNode(val, left, right) {
    this.val = (val===undefined ? 0 : val)
    this.left = (left===undefined ? null : left)
    this.right = (right===undefined ? null : right)
}

function inorderTraversal(root){

    const res = []

    function inorder(node) {
        if (!node) {
            return
        }
        inorder(node.left)
        res.push(node.val)
        inorder(node.right)
    }

    inorder(root)
    return res 

}

let root = new TreeNode(1)
let nodeA = new TreeNode(null)
let nodeB = new TreeNode(2)
let nodeC = new TreeNode(3)

root.left = nodeA
root.right = nodeB
nodeB.left = nodeC

let res = inorderTraversal(root)
res.shift()  //remove the element before root

console.log("answer to the first example is:", res)